Analysis/source/bbmutauAnalyzer.cxx

```diff
@@ -12,32 +12,61 @@ class bbmutauAnalyzer : public BaseEventAnalyzer {
 protected:
     virtual EventRegion DetermineEventRegion(EventInfoBase& eventInfoBase, EventCategory /*eventCategory*/) override
     {
-        static const std::vector<DiscriminatorWP> working_points = {
-            DiscriminatorWP::VLoose, DiscriminatorWP::Loose, DiscriminatorWP::Medium
+        //static const std::vector<DiscriminatorWP> working_points = {
+        //    DiscriminatorWP::VLoose, DiscriminatorWP::Loose, DiscriminatorWP::Medium
+        //};
+
+	static const std::map<DiscriminatorWP,double> working_points = {
+            {DiscriminatorWP::VVLoose,2.0}, {DiscriminatorWP::Medium,0.15}
         };
 
         const LepCandidate& muon = eventInfoBase.GetFirstLeg();
         const LepCandidate& tau = eventInfoBase.GetSecondLeg();
         
         EventRegion region;
 
-        if(ana_setup.mode == SignalMode::HTT || ana_setup.mode == SignalMode::HTT_sync){
+	if(ana_setup.mode != SignalMode::HH && ana_setup.mode != SignalMode::HTT_sync){
             double mt = analysis::Calculate_MT(muon.GetMomentum(),eventInfoBase.GetMET().GetMomentum());
             if(mt >= cuts::H_tautau_2016::mt) return EventRegion::Unknown();
+	    double m_tt_vis = (muon.GetMomentum() + tau.GetMomentum()).M();
+	    if(m_tt_vis <= 50) return EventRegion::Unknown();
+	    //cut for tau ID SF compatibility
+	    //double pzeta = analysis::Calculate_Pzeta(muon.GetMomentum(),tau.GetMomentum(),eventInfoBase.GetMET().GetMomentum());
+	    //if(pzeta <= -25) return EventRegion::Unknown();
         }
 
         const bool os = !ana_setup.apply_os_cut || muon.GetCharge() * tau.GetCharge() == -1;
         region.SetCharge(os);
 
-        for(auto wp = working_points.rbegin(); wp != working_points.rend(); ++wp) {
-            if(tau->Passed(TauIdDiscriminator::byIsolationMVArun2017v2DBoldDMwLT2017, *wp)) {
-                region.SetLowerIso(*wp);
+	TauIdDiscriminator tauIdDiscriminator = TauIdDiscriminator::byIsolationMVArun2017v2DBoldDMwLT2017;
+	if(ana_setup.mode == SignalMode::TauPOG_deepTauVsJet || ana_setup.mode == SignalMode::TauPOG_deepTauVsJet_full)
+		tauIdDiscriminator = TauIdDiscriminator::byDeepTau2017v1VSjet;
+	if(ana_setup.mode == SignalMode::TauPOG_dpfTau)
+		tauIdDiscriminator = TauIdDiscriminator::byDpfTau2016v0VSall;
+
+	if(!tau->Passed(tauIdDiscriminator, ana_setup.tauID_wp)) return EventRegion::Unknown();
```

Comment:
this selection should be done only when QCD is estimated by inverting muon isolation. Need to add an option to select which inversion logic should be followed.